### ISSUE TITLE
nativeGetSharedRealm crash

### ISSUE THREAD

**Issue by ch-vadim-maximov** [realm/realm-java#6468]
Hello getting some native crash reports by the Play dev console. Can't reproduce it, and seems like most of the users are fine.

Looks like it's happening here:
```
 RealmConfiguration config = new RealmConfiguration.Builder()
                .name(DB_FILE_NAME)
                .encryptionKey(key)
                .deleteRealmIfMigrationNeeded()
                .build();
        return Realm.getInstance(config);
```
Any ideas why this is happening would be greatly appreciated. 

```
#15  pc 000000000006f6c8  /data/app/[package-name-removed]-BIbCKYltmht3tcp9PvWvpg==/lib/arm64/librealm-jni.so (Java_io_realm_internal_OsSharedRealm_nativeGetSharedRealm+480)
  #16  pc 000000000003c824  /data/app/[package-name-removed]-BIbCKYltmht3tcp9PvWvpg==/oat/arm64/base.odex (io.realm.internal.OsObjectSchemaInfo.nativeGetProperty [DEDUPED]+180)
```
#### Version of Realm and tooling

Realm version(s): 5.7.1

Realm Sync feature enabled: No

Android Studio version: 3.3.2

Android Build Tools version: 28.0.3

Gradle version: 5.1

**Comment by ch-vadim-maximov** [comment]
Just read through the other issues, it's probably the same as: https://github.com/realm/realm-java/issues/6466

**Comment by cmelchior** [comment]
Hi @ch-vadim-maximov Can you provide the full stack trace? There should also be an exception message in clear text somewhere

**Comment by Zhuinden** [comment]
Is that a native stack overflow error? 🤔

**Comment by dd-david** [comment]
@Zhuinden I have a question about native stack overflow.

**Comment by cmelchior** [comment]
Since encryptions is enabled, I'm going to assume that this was caused by the bug fixed in Realm Java 6.0.1. Please try to upgrade. If that doesn't help feel free to re-open the issue.
